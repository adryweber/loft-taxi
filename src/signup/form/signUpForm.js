import React from "react";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { useStyles } from "../../shared/style";

export const SignUpForm = ({ signUp }) => {
    const classes = useStyles();

    const onSubmit = e => {
        // e.preventDefault();
    };

    return (
        <form onSubmit={onSubmit()}>
            <TextField fullWidth id="email" label="Эл. почта" margin="normal" />
            <TextField fullWidth id="firstname" label="Имя" margin="normal" />
            <TextField
                fullWidth
                id="lastname"
                label="Фамилия"
                margin="normal"
            />
            <TextField fullWidth id="password" label="Пароль" margin="normal" />
            <Button
                fullWidth
                className={classes.formBottomButton}
                variant="contained"
                color="primary"
            >
                Login
            </Button>
        </form>
    );
};
