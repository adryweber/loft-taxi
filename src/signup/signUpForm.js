import React from "react";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { useStyles } from "../shared/style";

export const SignUpForm = ({ setPage }) => {
    const classes = useStyles();

    const onSubmit = e => {
        e.preventDefault();
        setPage("profile");
    };

    return (
        <form>
            <TextField fullWidth id="email" label="Эл. почта" margin="normal" />
            <TextField fullWidth id="firstname" label="Имя" margin="normal" />
            <TextField
                fullWidth
                id="lastname"
                label="Фамилия"
                margin="normal"
            />
            <TextField fullWidth id="password" label="Пароль" margin="normal" />
            <Button
                fullWidth
                onClick={onSubmit}
                className={classes.formBottomButton}
                variant="contained"
                color="primary"
            >
                Login
            </Button>
        </form>
    );
};
