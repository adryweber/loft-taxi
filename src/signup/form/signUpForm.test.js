import React from "react";
import { SignUpForm } from "./signUpForm";
import renderer from "react-test-renderer";

// it("SignUpForm renders correctly", () => {
//     const tree = renderer.create(<SignUpForm />).toJSON();
//     expect(tree).toMatchSnapshot();
// });

it("SignUpForm renders correctly", () => {
    const tree = renderer
        .create(<SignUpForm signUp={() => ({ preventDefault: () => {} })} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
