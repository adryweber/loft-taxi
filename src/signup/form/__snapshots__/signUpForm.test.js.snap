// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm renders correctly 1`] = `
<form>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="email"
    >
      Эл. почта
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="email"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="firstname"
    >
      Имя
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="firstname"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="lastname"
    >
      Фамилия
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="lastname"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="password"
    >
      Пароль
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="password"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formBottomButton-5 MuiButton-containedPrimary MuiButton-fullWidth"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Login
    </span>
  </button>
</form>
`;
